### PR TITLE
pcache: adding bulk loading of keys

### LINKED ISSUE #7: pcache: bulk loading of keys
It will be good if pcache supports bulk loading of keys. That will give better a performance.
There is a snippet how code looks like without bulk loading:
``` kotlin
val locations = locationIds.map { cache.get(key(it)) }.filter { it != null } as List<CachedLocation>
```

getAll:
https://cloud.google.com/appengine/docs/standard/java/javadoc/com/google/appengine/api/memcache/MemcacheService#getAll-java.util.Collection-

### PR DESCRIPTION
**Pull request by StanimirIliev**
Fixes #7

### CODE REVIEW

File: clouway-pcache-client-gae/src/main/java/com/clouway/api/pcache/extensions/gae/GAECacheManager.java [hunk 1]
```diff
@@ -72,7 +75,21 @@ public Object get(String key) {
     return result;
   }
 
-  /**
+    @Override
+    public List<Object> getAll(List<String> keys) {
```
Comment (mlesikov) [inline clouway-pcache-client-gae/src/main/java/com/clouway/api/pcache/extensions/gae/GAECacheManager.java]: I think that we can make it a little bit better, we can return an Hits(HitResult) that could return and the keys that was not found
```kotlin
val hits = cache.getAll(keys)

if(hits.missed.isEmpty) {
 return hits.addAll(origin.get(hits.missed)).toList() 
} else {
 return hits.toList()
}
```
the names are not the best

other suggestion could be 
```kotlin
val hits = cache.getAll(keys, {missedKeys ->  origin.get(missedKeys) })
```
what do you think? 

maybe we can have 2 methods

File: clouway-pcache-client-testing/src/main/java/com/clouway/api/pcache/testing/CacheManagerContract.java [hunk 2]
```diff
@@ -124,6 +130,40 @@ public void flushCacheWhenThereIsNoItems() {
...
+
+    for(Object value : values) {
+        parsedValues.add(value.toString());
+    }
+
+    Assert.assertThat(parsedValues, is(equalTo(Collections.singletonList("::value1::"))));
+  }
+
+  @Test
+  public void getManyValuesWhenNonePersist() {
```
Comment (mlesikov) [inline clouway-pcache-client-testing/src/main/java/com/clouway/api/pcache/testing/CacheManagerContract.java]: noCachedValues

File: clouway-pcache-client-testing/src/main/java/com/clouway/api/pcache/testing/CacheManagerContract.java [hunk 3]
```diff
@@ -124,6 +130,40 @@ public void flushCacheWhenThereIsNoItems() {
...
+
+    for(Object value : values) {
+        parsedValues.add(value.toString());
+    }
+
+    Assert.assertThat(parsedValues, is(equalTo(Arrays.asList("::value1::", "::value2::"))));
+  }
+
+  @Test
+  public void getManyValuesWhenSomePersist() {
```
Comment (mlesikov) [inline clouway-pcache-client-testing/src/main/java/com/clouway/api/pcache/testing/CacheManagerContract.java]: partialCacheHits

File: clouway-pcache-client-gae/src/main/java/com/clouway/api/pcache/extensions/gae/GAECacheManager.java (hunk 1 above)
Comment (mgenov) [inline clouway-pcache-client-gae/src/main/java/com/clouway/api/pcache/extensions/gae/GAECacheManager.java]: You can return: <code>Map<String, Object></code> as the client would be able to check with O(1) speed which keys are returned and which not. 

The keys() and values() of Map are returned as set and sets are having easiest way for finding of union and diff. 

https://stackoverflow.com/questions/18644579/getting-the-difference-between-two-sets

Comment (StanimirIliev) [reply clouway-pcache-client-gae/src/main/java/com/clouway/api/pcache/extensions/gae/GAECacheManager.java]: Sounds good to me. Applied.

File: clouway-pcache-client-testing/src/main/java/com/clouway/api/pcache/testing/CacheManagerContract.java (hunk 2 above)
Comment (StanimirIliev) [reply clouway-pcache-client-testing/src/main/java/com/clouway/api/pcache/testing/CacheManagerContract.java]: This is done

File: clouway-pcache-client-testing/src/main/java/com/clouway/api/pcache/testing/CacheManagerContract.java (hunk 3 above)
Comment (StanimirIliev) [reply clouway-pcache-client-testing/src/main/java/com/clouway/api/pcache/testing/CacheManagerContract.java]: Done

**Review by mlesikov — APPROVED** [review]

**Pull request merged**